### DISCUSSION TITLE
Load a html page from public folder as root route

### DISCUSSION THREAD

**Discussion by FutureKode** [vercel/next.js#18189] [Help] [answered]
Hello,

We have a nextjs app for an online shop. We also have a complex animated welcome page built in static HTML and JS. Rather than converting the welcome page to React, I would like to load the HTML page from the public folder when user is at the root route '/'.

Anyone know how to do this?

Thanks

**Comment by jamesmosier** [comment, marked as answer]
The `public` folder is really just meant to serve static assets (i.e. images, documents, etc.). It'll also lose any of the optimizations from the Next build process. 

With that said, there is no mechanism currently to allow this. You would be better off hosting that static file elsewhere (i.e. outside of your Next app) and using a proxy to route to that file.

**Reply by FutureKode** [reply]
Thanks James, so you mean have a index.js under pages. And in that page do a fetch request for this external HTML file and then use dangerouslySetInnerHTML ?

**Reply by jamesmosier** [reply]
No, sorry. I meant you would be better off just convert the page to JSX and putting it in `pages/index.js`. You'll then gain all the great things/optimizations about Next 😄 

Your other option, if you didn't want to convert it to a React component, would be to serve it and route to it using your own proxy/router: for example using nginx or apache to route to the static file when a user goes to your homepage.
